benchmark/FedHPOB/fedhpob/utils/tabular_dataloader.py

```diff
@@ -96,6 +111,17 @@ def logs2df(dname,
                 x: []
                 for x in ['train_time', 'eval_time', 'tol_time']
             }
+            # Load fairness-related metric if exists.
+            fairness_list = None
+            fairness_gz = os.path.join(path, file_name, 'eval_results.raw.gz')
+            fairness_log = os.path.join(path, file_name, 'eval_results.raw')
+            if os.path.exists(fairness_gz):
+                with gzip.open(fairness_gz, 'rb') as f:
+                    fairness_list = read_fairness(f.readlines())
```

Comment:
Is it necessary to avoid loading all the content but reading a (large) file line by line as usual